### DISCUSSION TITLE
Duty cycle limits

### DISCUSSION THREAD

**Discussion by faragher** [markqvist/Reticulum#273] [Requests for New Features]
I've been going through the datasheets recently, and a thought occurs to me: I have a vague recollection of the US limiting duty cycle on data transmission over certain bands. As a practical matter, this doesn't make much difference at the scales most of use use these on, and I don't remember the exact bands and modes affected.

However, I think a duty cycle option is useful. More research is required to figure out the specifics, but I think if we have a duty cycle as a fraction (defaults to one) with a window (default to, say, 60 seconds) and a computed length of message or simple timer, we can leave the actual configuration and compliance up to the end user. I would propose it be part of the Reticulum stack instead of the RNodes, as it is application specific. Alternatively, this could be handled by each system, such as Nomadnet or Sideband, but it's a feature that would be required for widespread compliance and acceptance.

Is this something you think should be implemented?

**Comment by markqvist** [comment]
It's been on my todo list for way too long. As you say, the amount of traffic is so low with the current functionalities that it is more or less self-enforcing. That being said, I think it should be an option, and I'm going to add it at some point, when I have a bit of time for it ;)

**Comment by LinuxinaBit** [comment]
Going to bump this.
iirc it’s not necessary in the EU because of RNode’s ‘advanced interference mitigation’, but it still might be necessary in some other places.

**Reply by markqvist** [reply]
Already there :)

The duty cycle (airtime) limiting has been in the firmware since mid 2023 (as far as I remember). It was implemented shortly after this post was made. 

It can be configured both directly in the Sideband RNode settings on Android, the Reticulum interfaces configuration, or any other program that supports all the interface options that Reticulum exposes.

**Reply by markqvist** [reply]
There's even options to set both a short-term (15 seconds) and a long-term (1 hour) limit individually. All places I am aware of specify the limits over 1 hour periods.

**Reply by LinuxinaBit** [reply]
Ooh, must've missed it!
That's fantastic :heart:
